### DISCUSSION TITLE
DI

### DISCUSSION THREAD

**Discussion by IYoni** [doppelganger9/babyfoot#771] [Ideas]
As I see it, your application can be one of the benchmarks for someone to sprint from, it is a great artifact to take it, learn it, and use it for developments. I would like to contribute in your project, and I think we can add a DI into, I believe it will help support modularization. Please, put your comment, how we can start off, if DI is not part of your project already?

**Comment by doppelganger9** [comment]
Hi there,

By all means, fork this projects and use it as you wish!

This project is based on someone else's work I saw in a conference, it was written in C#/dotNET so as/ an exercise I rewrote it using Typescript/NodeJS. I wanted to practice event-driven CQRS/ES to get a feel if I could promote it at work.

If by "DI" you mean Dependency Injection, if I remember correctly, I tried to avoid coupling the code to any specific tool, and kept things simple using Constructors. 
The main application is responsible for wiring Objects using constructors.
Constructor Injection eases writing unit tests: you create an instance of your tested object via its constructor, injecting mocks for its collaborators.

When you say modularization what do you had in mind?

**Reply by IYoni** [reply]
Yes exactly I was talking about Dependency Injection, and I was referring to a containerized initialization that would be modularization. 
Hmm...lesson learned, you rewrote your app from seeing someone else's work and that is even written in a different language...that takes a real passion, I really appreciate you. thank you.

**Comment by IYoni** [comment]
I actually better think modularization in terms of monorepo. You can forget me, as you pointed out better to keep it simple.
